lib/curl-command-builder.js

```diff
@@ -0,0 +1,236 @@
+ * @param  {array<string>} methodParams.headers   An array of serialized HTTP headers
+ * @param  {*}             methodParams.payload   The payload to send in the request
+ * @param  {array<object>} methodParams.securedBy An array of security references from the RAML
+ * @param  {object}        securitySchemes        A hash of Security Scheme objects as generated by RAML
+ * @return {array<string>}
+ */
+function forMethod (methodParams, securitySchemes) {
+  let commands = []
+  if (securitySchemes && methodParams.securedBy) {
+    commands = securedCommands(methodParams, securitySchemes)
+  }
+  if (commands.length === 0) {
+    commands = unsecuredCommands(methodParams)
+  }
+  return commands
+}
+
+/**
+ * Builds a curl command string without security
+ * @param {object} methodParams See definition in forMethod
+ */
+function unsecuredCommands (methodParams) {
+  return buildCurlCommands(methodParams)
+}
+
+/**
+ * Builds curl commands with security schemes
+ * @param {object} methodParams     See definition in forMethod
+ * @param {object} securitySchemes  A hash of Security Scheme objects as generated by RAML
+ * @return {array<string>}
+ */
+function securedCommands (methodParams, securitySchemes) {
+  let commands = []
+  methodParams.securedBy.forEach((methodScheme) => {
+    if (methodScheme) {
+      let securityScheme = securitySchemes[methodScheme.schemeName]
+      if (securityScheme) {
+        commands = commands.concat(buildCurlCommands(methodParams, securityScheme))
+      }
+    } else {
+      commands = unsecuredCommands(methodParams)
+    }
+  })
+  return commands
+}
+
+/**
+ * Builds one or more curl commands for a method and set of security schemes
+ * @param  {object}        methodParams         The set of parameters describing a methof
+ * @param  {string}        methodParams.method  The HTTP method
+ * @param  {string}        methodParams.baseUri The base URI
+ * @param  {string}        methodParams.path    The path
+ * @param  {array<string>} methodParams.params  An array of serialized key-value pairs that make up the query string parameters
+ * @param  {array<string>} methodParams.headers An array of sewrialized HTTP headers
+ * @param  {*}             methodParams.payload The payload to send in the request
+ * @param  {object}        securityScheme       A Security Scheme object as generated by RAML
+ * @return {array<string>}                      An array of strings that represents the possible curl commands
+ */
+function buildCurlCommands (methodParams, securityScheme) {
+  const curlAuthFunctions = {
+    'Basic Authentication': curlBasicAuth,
+    'Digest Authentication': curlDigestAuth,
+    'Pass Through': curlPassThroughAuth,
+    'x-custom': curlXCustomAuth,
+    'OAuth 1.0': curlOAuth1,
+    'OAuth 2.0': curlOAuth2,
+    'null': curlNullAuth
+  }
+
+  securityScheme = securityScheme || {type: 'null'}
+  const authFunction = curlAuthFunctions[securityScheme.type] || curlNullAuth
```

Comment:
The spec allows for `x-{other}` as a custom type, so `x-custom` is not sufficient.
I suppose we could do a pipeline that shortcircuits:
```js
const authFunction = curlAuthFunctions[securityScheme.type] || curlCustomAuth(securityScheme.type) || curlNullAuth

function curlCustomAuth(type) {
  return type.startsWith('x-') ? curlXCustomAuth : false
}
```
Or maybe wrap the pipeline into a single function

Reply:
Oh, interesting. I read the spec as saying that the value for `type` will be `x-custom`: https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md#x-other